clients/python/src/model_registry/_client.py

```diff
@@ -437,3 +442,228 @@ def rm_versions(options: ListOptions) -> list[ModelVersion]:
             return self.async_runner(self._api.get_model_versions(rm.id, options))
 
         return Pager[ModelVersion](rm_versions)
+
+    def save_to_s3(
+        self,
+        path: str,
+        bucket_name: str,
+        *,
+        s3_prefix: str | None = None,
+        endpoint_url: str | None = None,
+        access_key_id: str | None = None,
+        secret_access_key: str | None = None,
+        region: str | None = None,
+    ) -> str:
+        """Saves a model to an S3 compatible storage.
+
+        Args:
+            path: Location to where the model(s) or artifact(s) are located. \
+                Can recursively upload nested files in folders.
+            bucket_name: The bucket to use for the S3 compatible object storage.
+
+        Keyword Args:
+            s3_prefix: The path to prefix with under root of bucket. If unset, will default to `path` if \
+                `path` is not a nested folder.
+            endpoint_url: The endpoint URL for the S3 comaptible storage if not using AWS S3.
+            access_key_id: The S3 compatible object storage access ID.
+            secret_access_key: The S3 compatible object storage secret access key.
+            region: The region name for the S3 object storage.
+
+        Returns:
+            S3 URI formatted from the object
+
+        Raises:
+            StoreError: If there was an issue uploading to S3.
+        """
+        # Get mixed credentials
+        endpoint_url, access_key_id, secret_access_key, region = self.__s3_creds(
+            endpoint_url, access_key_id, secret_access_key, region
+        )
+
+        s3 = self.__connect_to_s3(
+            endpoint_url=endpoint_url,
+            access_key_id=access_key_id,
+            secret_access_key=secret_access_key,
+        )
+        try:
+            return self.__upload_to_s3(
```

Comment:
just to double-check: in the case of a folder, this will return an array list, and we will need to determine from the other PR here:
- https://github.com/kubeflow/model-registry/pull/761
the _single_ URI to register in MR, to be later used to populate the Isvc.
Correct?
cc @Crazyglue